frontend/src/pages/pipelines/global/modelCustomization/HyperparameterSection/HyperparameterFields.tsx

```diff
@@ -0,0 +1,91 @@
+import { Stack } from '@patternfly/react-core';
+import * as React from 'react';
+import { HyperparametersFormData } from '~/concepts/pipelines/content/modelCustomizationForm/modelCustomizationFormSchema/validationUtils';
+import {
+  HYPERPARAMETER_EVALUATION_LIST,
+  HYPERPARAMETER_LONG_NUMBER_LIST,
+  HyperparameterFields,
+} from '~/pages/pipelines/global/modelCustomization/const';
+import { asEnumMember } from '~/utilities/utils';
+import ParamsDefaultFields from '~/components/ParamsDefaultFields';
+import { HYPERPARAMETER_MAP } from './HyperparameterFieldsUtils';
+
+type HyperparameterFieldsDisplayProps = {
+  hyperparameters: HyperparametersFormData;
+  isEmpty: boolean;
+  onChange: (
+    hyperparameter: HyperparameterFields,
+    hyperparameterValue: string | number | boolean,
+  ) => void;
+};
+
+const HyperparameterFieldsDisplay: React.FC<HyperparameterFieldsDisplayProps> = ({
+  hyperparameters,
+  isEmpty,
+  onChange,
+}) => (
+  <Stack hasGutter>
+    {!isEmpty &&
+      Object.entries(hyperparameters).map(([key, value]) => {
+        const hyperparameterKey = asEnumMember(key, HyperparameterFields);
+        if (hyperparameterKey) {
+          const { label, renderField } = HYPERPARAMETER_MAP[hyperparameterKey];
+          if (renderField) {
+            if (HYPERPARAMETER_LONG_NUMBER_LIST.includes(hyperparameterKey)) {
```

Comment:
This is good, but we should not be iterating over the form data, but a separate piece of data passed in called something like hyperparameterSpec which should be an array of ParameterKF directly taken from the pipeline and gone through your filtering function. Then we can check if its known (in your HYPERPARAMETER_MAP) and render it using your renderField. Otherwise use you ParamsDefaultFields